### ISSUE TITLE
Error when creating factor with int variable names

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#387]
``` python
phi = Factor([3, 2], [2, 2], [0.39, 0.06, 0.61, 0.94])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-8c522fe50ddd> in <module>()
----> 1 phi = Factor([3, 2], [2, 2], [0.39, 0.06, 0.61, 0.94])

/home/ankur/pgmpy/pgmpy/factors/Factor.py in __init__(self, variables, cardinality, value)
     73         for variable, card in zip(variables, cardinality):
     74             self.variables[variable] = [variable + '_' + str(index)
---> 75                                         for index in range(card)]
     76         self.cardinality = np.array(cardinality)
     77         self.values = np.array(value, dtype=np.double)

/home/ankur/pgmpy/pgmpy/factors/Factor.py in <listcomp>(.0)
     73         for variable, card in zip(variables, cardinality):
     74             self.variables[variable] = [variable + '_' + str(index)
---> 75                                         for index in range(card)]
     76         self.cardinality = np.array(cardinality)
     77         self.values = np.array(value, dtype=np.double)
```

**Comment by palashahuja** [comment]
@ankurankan Quickfix : use the `str` definition. That will fix it

**Comment by ankurankan** [comment]
#392 should fix this.

**Comment by ankurankan** [comment]
Fixed in #407
